src/Settings/ProjectSettings.ts

```diff
@@ -23,10 +24,30 @@ export class ProjectSettings extends Settings {
     }
 
     /**
-     * Gets a value by key from Azure DevOps.
+     * Gets the value of the given key from the given map of existing Azure DevOps project settings. If the key
+     * does not exist, then the given default value is returned.
      */
-    public get(key: string): PromiseLike<string> {
-        return super.get(this.getProjectSpecificKey(key));
+    public getOrDefault(key: string, existingSettings : Map<string, string>, defaultValue: string): string {
+        const settingKey = this.getProjectSpecificKey(key);
+        if(existingSettings.has(settingKey)){
+            return existingSettings.get(settingKey);
+        }
+        return defaultValue;
```

Comment:
thought: I know this was already the case before your change, but I don't really like that we have the default value as parameter here. I'd prefer having it as part of the constants defined in `Settings` or similar (by changing their type to an enum/object/...). But I'm not sure if this refactoring would be way to large for this ticket...

Reply:
Added new object/class `ExtensionSetting` which holds all the settings for the Teamscale extension along with their default values.